src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -372,6 +372,39 @@ public void shutdownCommissioning() {
   private native PaseVerifierParams computePaseVerifier(
       long deviceControllerPtr, long devicePtr, long setupPincode, long iterations, byte[] salt);
 
+  /**
+   * Sets the key configuration used for operational control of devices. This includes all necessary
```

Comment:
Why are we adding this mutator API instead of just having consumers pass the right information to initial controller bring-up?

This API is _extremely_ unsafe and hard to use properly.  If we can avoid adding it, we should.

Reply:
Can you provide some context as to how this is unsafe, vs providing it at construction time? Having a mutator allows us to change our operational context dynamically. We could refactor and just create separate instances per operational context but it's not clear why that is preferable here.

Additionally, in some cases there may not be an operational context (e.g. commissioning the first device into a fabric).

Reply:
You can't change the fabric of a controller while it's actually operating.  It's got existing CASE sessions open, for example, and suddenly its identity is supposed to change.  Or operational advertising: that will not update when this state gets changed via this API.

If you read the documentation for the API (which was added, as far as I can tell, solely for this caller), it very clearly says that you have to be _extremely_ careful about when you use it.  The only safe time to use it is immediately after creating the object.

> Additionally, in some cases there may not be an operational context (e.g. commissioning the first device into a fabric).

This API is all about self-commissioning a controller/commissioner onto a fabric, no?  The "right" way to do that is to make sure that you have all the crypto material needed, _then_ create the object.

What's happening now is that the commissioner is started on some random fabric with some random crypto material (presumably; you have to provide certs and the like to create a DeviceController, and I assume some placeholder ones are being provided), then this API changes the crypto material without notifying anything else in the system about it.